### DISCUSSION TITLE
Why not Next JS?

### DISCUSSION THREAD

**Discussion by sandheep45** [prasenjeet-symon/intellectia#80] [General]
I am curious to know why you are opting Vite instead of Next JS as if you visit react official docs, they are suggesting to use feamework like Next or Remix over standalone React

**Comment by prasenjeet-symon** [comment]
Truthfully, our primary goal in this decision is to enhance the development efficiency of our application, `APP DIR` has been experiencing performance challenges in its current state. Furthermore, we aim to broaden the accessibility of this project to a wider community of potential contributors, many of whom possess extensive experience with vanilla React.

Introducing Next.js, while undoubtedly powerful, would introduce additional complexities in managing both server and client components, potentially complicating the mental models of developers who are already well-versed in vanilla React. Our intention is to maintain a development environment that aligns with the expertise and familiarity of our existing contributors.

**Reply by sandheep45** [reply]
Thank you for explaining me your vision

**Comment by mirislomovmirjalol** [comment]
### There are a few reasons why I might opt for Vite over Next.JS:

- Performance: Vite is a very fast build tool and development server, thanks to its use of native ES modules and other optimizations. This can make a big difference in the development experience, especially for large projects.
- Flexibility: Vite is more flexible than Next.JS, as it is not tied to any particular framework or library. This makes it a good choice for projects that need to use a mix of different technologies.
- Simplicity: Vite is simpler to understand and use than Next.JS. This makes it a good choice for beginners or for projects that do not need all of the features that Next.JS has to offer.

### However, there are also some advantages to Next.JS:

- Features: Next.JS offers a number of features that are not available in Vite, such as server-side rendering, static site generation, and built-in support for authentication and authorization.
- Maturity: Next.JS is a more mature framework than Vite, with a larger community and ecosystem. This means that there are more resources available to help you get started and solve problems.
